From cf02487863abdf193adcff8178874a0144231004 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Wed, 14 Dec 2016 16:00:40 +0530
Subject: [PATCH 028/205] ARM: OMAP2+: omap_hwmod: Always restore saved
 hardreset context

This commit comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

Previously when restoring hardreset context during
omap_hwmod_restore_context we would only deassert the hardreset lines if
the module was previously active, however, if a hwmod has all hardresets
asserted then _enable will return without actually enabling the module.

This is a problem for the gfx hwmod on am437x as it gets disabled in
suspend path so it appears as disabled to the restore context code but
then during the attempted enable call during the regular kernel resume
path, the hwmod cannot actually be enabled.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
(cherry picked from commit b67ad14b10d091e4e4f3cc339deb88f438439b42)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/omap_hwmod.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod.c b/arch/arm/mach-omap2/omap_hwmod.c
index 1fd6d64..e7c2135 100644
--- a/arch/arm/mach-omap2/omap_hwmod.c
+++ b/arch/arm/mach-omap2/omap_hwmod.c
@@ -3920,7 +3920,7 @@ static int omap_hwmod_restore_context(struct omap_hwmod *oh, void *unused)
 	for (i = 0; i < oh->rst_lines_cnt; i++)
 		if (oh->rst_lines[i].context)
 			_assert_hardreset(oh, oh->rst_lines[i].name);
-		else if (oh->_state == _HWMOD_STATE_ENABLED)
+		else
 			_deassert_hardreset(oh, oh->rst_lines[i].name);
 
 	if (oh->_state == _HWMOD_STATE_ENABLED) {
-- 
1.7.9.5

